### PR TITLE
Refactor Redirect Delay

### PR DESCRIPTION
**Pull request by alsoicode**
## Rationale
djangocms-forms uses a hard-coded redirect delay value of 1000, which may not suit all use cases.

This pull request makes the redirect delay value configurable on a per-form basis via the FormDefinition model, site-wide via settings or falling back to the original value of 1000 milliseconds.

## Work
- Added `redirect_delay` field to FormDefinition model.
- Added migrations for `redirect_delay` field for Django 1.8+ and South.
- Updated cms plugin to include `redirect_delay` in fieldsets and pass `redirect_delay` value to template.
- Updated `FormDefinitionAdminForm` to ensure a redirect is provided if a `redirect_delay` is specified.
- Updated default form template to pass `redirect_delay` to the `cmsForms` object constructor.
- Updated cmsForms JavaScript to pass in `redirect_delay` value and use the setting value instead of a hard-coded value of 1000 milliseconds.
- Version bump to 0.1.16.
- Updated documentation on use of `redirect_delay` field.

## Results
test form - google chrome_014

#### per-form redirect delay
- Create a form, and enter either a page or external site to redirect to as well as a `redirect_delay` value.
- Publish your changes.
- Visit the page with your form and submit it.
- Verify that the correct number of milliseconds has passed before the redirect occurs.

#### site-wide redirect delay
- Add a `DJANGOCMS_FORMS_REDIRECT_DELAY` to your settings.
- Using Google Chrome Developer tools or other DOM inspector, verify that the number of milliseconds is displayed for the `redirect_delay` (see screenshot above for example)
- Create a form. Do not enter a `redirect_delay` value. Publish.
- Visit the page with your form and submit it.
- Verify that the correct number of milliseconds has passed before the redirect occurs.

#### default value
- Remove the `DJANGOCMS_FORMS_REDIRECT_DELAY` from your settings.
- Create a form. Do not enter a `redirect_delay` value. Publish.
- Visit the page with your form and submit it.
- Verify that the correct number of milliseconds has passed before the redirect occurs.

### CODE REVIEW

**Comment by mishbahr** [comment]
Thank you so much for this pull request. I'm super busy for the next few days.

I'll merge this is as soon as I can.

Thanks

Mishbah

**Comment by alsoicode** [comment]
You're very welcome, and thanks.

File: djangocms_forms/templates/djangocms_forms/form_template/default.html [hunk 1]
```diff
@@ -63,8 +63,10 @@ <h3 class="title">{{ instance.title }}</h3>
 
 	<script type="application/javascript">
 		$(function() {
-		  $('.forms').cmsForms({'reCaptchaSiteKey': '{{ recaptcha_site_key }}'});
+		  $('.forms').cmsForms({
+			  'reCaptchaSiteKey': '{{ recaptcha_site_key }}',
+			  'redirect_delay': {{ redirect_delay|default:1000 }}
```
Comment (mishbahr) [inline djangocms_forms/templates/djangocms_forms/form_template/default.html]: This should be `redirectDelay`?

Comment (alsoicode) [inline djangocms_forms/templates/djangocms_forms/form_template/default.html]: Good catch. Yes.

**Pull request merged**